### PR TITLE
feat: Create or update version

### PR DESCRIPTION
**Pull request by tienvx**
I have exactly the `chicken & egg` problem which is described at https://github.com/pact-foundation/pact_broker-client/issues/25#issue-287038484

After asking in `pact-broker` channel, the solution is to run `create-or-update-version` at the beginning of the workflow/pipeline. https://pact-foundation.slack.com/archives/C9VPNUJR2/p1689584709582369

There is no official Github action for it, so I created for it.

### CODE REVIEW

File: create-or-update-version/createOrUpdateVersion.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+"
+
+docker run --rm \
+    -e PACT_BROKER_BASE_URL=$PACT_BROKER_BASE_URL \
+    -e PACT_BROKER_TOKEN=$PACT_BROKER_TOKEN \
+    pactfoundation/pact-cli:latest \
+    broker create-or-update-version \
+    --pacticipant "$application_name" \
+    --version $version \
+    --branch $branch
```
Comment (YOU54F) [inline create-or-update-version/createOrUpdateVersion.sh:32]: is branch required? if so it should be a workflow input i believe, but would need to check the others

Comment (YOU54F) [reply create-or-update-version/createOrUpdateVersion.sh:32]: actually it should be --main-branch

https://github.com/pact-foundation/pact_broker-client#create-or-update-pacticipant

**Review by YOU54F — CHANGES_REQUESTED** [review]
can you change branch to be main-branch as per the docs

https://github.com/pact-foundation/pact_broker-client#create-or-update-pacticipant

i think the failure in the github actions run is due to token permissions, it’s not related to your change, sorry about that!

thanks for adding tests too :)

will review this on the laptop tomorrow and look to get it merged for you

**Comment by YOU54F** [comment]
thanks for the pr dude!

File: create-or-update-version/createOrUpdateVersion.sh (hunk 1 above)
Comment (tienvx) [reply create-or-update-version/createOrUpdateVersion.sh:32]: It's `--branch`. You are looking at the other link. It should be https://github.com/pact-foundation/pact_broker-client#create-or-update-version

I tested by running this command on my own project and it look like `--branch` is not required

```shell
docker run --rm \                                                                                                                                                                                                                                                                              09:17:09
                               -e PACT_BROKER_BASE_URL=https://my-project.pactflow.io \
                               -e PACT_BROKER_TOKEN=MYtoken \
                               pactfoundation/pact-cli:latest \
                               broker create-or-update-version \
                               --pacticipant "name" \
                               --version 123
Created/updated pacticipant version 123 in PactFlow
```

Comment (tienvx) [reply create-or-update-version/createOrUpdateVersion.sh:32]: And yes, I'm not sure what is the best way to handle `--branch` there:
* Keep it like [publish-pact-files](https://github.com/pactflow/actions/blob/main/publish-pact-files/publishPactfiles.sh#L15)?
* Remove it completely?
* Convert it to input to allow user to change it?

Comment (YOU54F) [reply create-or-update-version/createOrUpdateVersion.sh:32]: hmmm sorry my bad in reading the wrong docs, thanks for clarfication.

not sure if someone would want to create a pacticipant version with a branch every time? 

If yes, they setting it as per `publish-pact-files` might be good, so it is automatically determined and applied.

I wonder if a user may choose to only run this on certain branches (which they can stop via an if condition to run only if x branch) when executing the action.

maybe 
 - apply the branch by default as per [publish-pact-files](https://github.com/pactflow/actions/blob/main/publish-pact-files/publishPactfiles.sh#L15)
- allow user to override branch via branch input
- allow user to disable branch association (apply_branch=false, with the default value of true)

That might cover the cases where a user doesn't want to add a branch.

Comment (tienvx) [reply create-or-update-version/createOrUpdateVersion.sh:32]: So I think let keep it like [publish-pact-files](https://github.com/pactflow/actions/blob/main/publish-pact-files/publishPactfiles.sh#L15) for now

Comment (YOU54F) [reply create-or-update-version/createOrUpdateVersion.sh:32]: yeah that makes sense to me :)

**Pull request merged**